wordpress-mobile/WordPress-Android
# [Modules Restructuring] Restructure Editor Module

Parent #16562
Fixes #16564

This PR restructures the `editor` module by disassociating the [WordPress-Editor-Android](https://github.com/wordpress-mobile/WordPress-Editor-Android) repo and removing the extra `WordPressEditor` level from it.  This effectively makes `editor` a pure module, without any extra inner levels or unnecessary build files lying around (like `settings.gradle`, `.gitignore`, etc).

Also, this PR activates the `editor-example` module and placing that on the same level as the `editor` module.

PS.1: @antonis I added you as the main reviewer, that is, in addition to the @wordpress-mobile/owl-team team itself, since you helped us with the [OPA Android](https://github.com/Automattic/opa-android) project a while back, which involved working the `editor` and testing it, starting with the `editor-example`. As such, I think you might be the best person to sign-off on that change for WPAndroid. 🥇

PS.2 @antonis as part of the above, please take a moment to also sign-off the `editor-example` module changes as well (especially [here](https://github.com/wordpress-mobile/WordPress-Android/pull/16769/commits/054b97523060cb0613714eae32d0867943955dd3) and [here](https://github.com/wordpress-mobile/WordPress-Android/pull/16769/commits/bad8f188fa0bf5e85be459c30d18916437129935)). With this change you will notice that this `editor-example` module is now active and buildable too, still, it fails to compile as there are `21` errors with the code itself. My intention with this PR is to ignore those compilation errors for now and maybe come back to fixing them later on with a separate PR, so don't worry too much about them, I'll add a task on it within the [parent](https://github.com/wordpress-mobile/WordPress-Android/issues/16562) ticket afterwards. You can try assembling this module by running the `./gradlew libs:editor-example:assembleDebug` command, but FYI, this doesn't affect building the main WPAndroid app anyway. Thus, it failing doesn't matter much, that is in terms of progressing this PR. Also, note that our CI or any of our processes are unaware of it, this is just a nice to have at this point.

PS.3: @AliSoftware I added you as a secondary reviewer, that is, in addition to the @wordpress-mobile/owl-team itself, as I did another [localization.rb](https://github.com/wordpress-mobile/WordPress-Android/pull/16769/commits/61a20737c220914d61cf8bbab03134c519f81030) related change that I want you to double check and verify again, just in case I missed something vital. Also, in addition to that, please double check this minor [strings.xml](https://github.com/wordpress-mobile/WordPress-Android/pull/16769/commits/edd52f2330bf9e13f108ea2d8bf978fdfac3598d) related change that update the string group section comment.

-----

Question (❓): As per my licensing question on this other related PR of mine [here](https://github.com/wordpress-mobile/WordPress-Android/pull/16744) and since I saw that the `LICENCE` file for this module is `GPLv2`, unlike the `analytics` mode, which is pointing to `MIT`, I deleted the licence for this module to start depending solely on the root level `LICENCE` that WPAndroid is using, which is `GPLv2` as well, wdyt? 🤔

-----

To test:
- Build the app and install it on your device or emulator.
- Go to `Posts` -> Click on the plus 🔵 FAB button on the bottom right -> Select `Blog post` to create a new post.
- Randomly pick and add few blocks to the post and verify that the post editing/publishing works as expected.

The above is just one example of how this change can tested so that we could quickly verify the overall correctness of this PR. However, feel free to smoke test other aspects as well, especially if you are aware any or all the other places where the `editor` is being used like `Pages` or `Stories`.

-----

## Regression Notes
1. Potential unintended areas of impact

N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:
